controllers/configurationpolicy_controller.go

```diff
@@ -1110,28 +1125,75 @@ func (r *ConfigurationPolicyReconciler) handleObjectTemplates(plc policyv1.Confi
 				relatedObjects = updateRelatedObjectsStatus(relatedObjects, object)
 			}
 		}
-		// violations for enforce configurationpolicies are already handled in handleObjects,
-		// so we only need to generate a violation if the remediationAction is set to inform
-		if !handled && !enforce {
-			objData := map[string]interface{}{
-				"indx":        indx,
-				"kind":        kind,
-				"desiredName": templateObjs[indx].name,
-				"namespaced":  templateObjs[indx].isNamespaced,
-			}
 
-			statusUpdateNeeded := createInformStatus(
+		objData := map[string]interface{}{
+			"indx":        indx,
+			"kind":        kind,
+			"desiredName": templateObjs[indx].name,
+			"namespaced":  templateObjs[indx].isNamespaced,
+		}
+		if mergeMessageEnforce || (!handled && !enforce) {
+			statusUpdateNeeded := createMergedStatus(
 				objShouldExist, numCompliant, numNonCompliant, compliantObjects, nonCompliantObjects, &plc, objData,
 			)
+
 			if statusUpdateNeeded {
 				parentStatusUpdateNeeded = true
 			}
+
+			if mergeMessageEnforce {
+				r.Recorder.Event(&plc, eventNormal, fmt.Sprintf(plcFmtStr, plc.GetName()),
+					convertPolicyStatusToString(&plc))
+			}
 		}
 	}
 
 	r.checkRelatedAndUpdate(plc, relatedObjects, oldRelated, parentStatusUpdateNeeded)
 }
 
+// createMergedStatus updates the status field for a configurationpolicy with remediationAction=inform
+// based on how many compliant/noncompliant objects are found when processing the templates in the configurationpolicy
+// Or multiple namespaces and enforce use this
+func createMergedStatus(
```

Comment:
Could you please keep this function where it was and just rename it and add the comment? That way the Git diff doesn't make it seem like you wrote a new function.